tests/components/whirlpool/test_climate.py

```diff
@@ -0,0 +1,343 @@
+    assert state.attributes.get(ATTR_CURRENT_TEMPERATURE) == 16
+    assert state.attributes.get(ATTR_TEMPERATURE) == 21
+    assert state.attributes.get(ATTR_CURRENT_HUMIDITY) == 70
+    assert state.attributes.get(ATTR_SWING_MODE) == SWING_OFF
+
+
+async def test_service_calls(hass: HomeAssistant, mock_aircon_api: MagicMock):
+    """Test controlling the entity through service calls."""
+    await init_integration(hass)
+    mock_aircon_api.return_value.set_power_on = AsyncMock()
+    mock_aircon_api.return_value.set_mode = AsyncMock()
+    mock_aircon_api.return_value.set_temp = AsyncMock()
+    mock_aircon_api.return_value.set_humidity = AsyncMock()
+    mock_aircon_api.return_value.set_mode = AsyncMock()
+    mock_aircon_api.return_value.set_fanspeed = AsyncMock()
+    mock_aircon_api.return_value.set_h_louver_swing = AsyncMock()
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "climate.said1"},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: "climate.said1"},
+        blocking=True,
+    )
+    mock_aircon_api.return_value.set_power_on.assert_has_calls(
+        [mock.call(False), mock.call(True)]
+    )
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "climate.said1"},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_COOL},
+        blocking=True,
+    )
+    mock_aircon_api.return_value.set_power_on.assert_has_calls(
+        [mock.call(False), mock.call(True)]
+    )
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_TEMPERATURE: 15},
+        blocking=True,
+    )
+    mock_aircon_api.return_value.set_temp.assert_called_once_with(15)
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_COOL},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_HEAT},
+        blocking=True,
+    )
+    # HVAC_MODE_DRY should be ignored
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_DRY},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_FAN_ONLY},
+        blocking=True,
+    )
+    mock_aircon_api.return_value.set_mode.assert_has_calls(
```

Comment:
Assert after each call to ensure each call works as expected instead of the end result.

Reply:
Good tip! There was indeed a bug on that part of the test!

Reply:
Good tip! There was indeed a bug on that part of the test!